### ISSUE TITLE
Exception from Auth is not catched by exception handler.

### ISSUE THREAD

**Issue by AntonMelentiev** [vitalik/django-ninja#296]
Hello!

I've tried to raise a custom exception from the authorization class and added an exception handler for it. But seems like this exception is not handled by the exception handler, just raised.

Example from documentation
https://django-ninja.rest-framework.com/tutorial/authentication/#custom-exceptions
also doesn't work for me. 
If I copy it to my app and send a request with some Bearer token that != to "supersecret" exception handler doesn't catch the error and I just get a response with status-code 500 and exception raised on the backend side:

```
Internal Server Error: /api/v1/bearer
Traceback (most recent call last):
  File "/home/anton/projects/project_name/.venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/anton/projects/project_name/.venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/anton/projects/project_name/.venv/lib/python3.8/site-packages/ninja/operation.py", line 311, in _sync_view
    return operation.run(request, *a, **kw)  # type: ignore
  File "/home/anton/projects/project_name/.venv/lib/python3.8/site-packages/ninja/operation.py", line 88, in run
    error = self._run_checks(request)
  File "/home/anton/projects/project_name/.venv/lib/python3.8/site-packages/ninja/operation.py", line 120, in _run_checks
    error = self._run_authentication(request)
  File "/home/anton/projects/project_name/.venv/lib/python3.8/site-packages/ninja/operation.py", line 134, in _run_authentication
    result = callback(request)
  File "/home/anton/projects/project_name/.venv/lib/python3.8/site-packages/ninja/security/http.py", line 39, in __call__
    return self.authenticate(request, token)
  File "/home/anton/projects/project_name/backend/api.py", line 152, in authenticate
    raise InvalidToken
backend.api.InvalidToken
```

**Comment by vitalik** [comment]
@AntonMelentiev 

could you show your code ? (`.exception_handler` and authenticator function )

**Comment by AntonMelentiev** [comment]
@vitalik for this error removed all my code and just copied it from the doc: https://django-ninja.rest-framework.com/tutorial/authentication/#custom-exceptions

```
from ninja import NinjaAPI
from ninja.security import HttpBearer

api = NinjaAPI()

class InvalidToken(Exception):
    pass

@api.exception_handler(InvalidToken)
def on_invalid_token(request, exc):
    return api.create_response(request, {"detail": "Invalid token supplied"}, status=401)

class AuthBearer(HttpBearer):
    def authenticate(self, request, token):
        if token == "supersecret":
            return token
        raise InvalidToken


@api.get("/bearer", auth=AuthBearer())
def bearer(request):
    return {"token": request.auth}
```

**Comment by vitalik** [comment]
@AntonMelentiev  

works for me:


```
curl -X 'GET' \
>   'http://localhost:8000/api/bearer' \
>   -H 'accept: */*' \
>   -H 'Authorization: Bearer invalidtoken'
{"detail": "Invalid token supplied"}
```

My guess is you are using some older version - auth execptions were implemented in #174 
so I guess you need to have at least `django-ninja=>0.14.0`

**Comment by AntonMelentiev** [comment]
Yep, you are right. My bad.
`django-ninja = "^0.13.2"`
Thanks for your support!
